engine/title.asm

```diff
@@ -3,17 +3,654 @@ INCLUDE "constants.asm"
+	ld [hl], e
+	inc hl
+	ld [hl], d
+	ret	
+	
+TitleScreenInputAndTimeout:: ;6114
+	ld hl, wSequence+2
+	ld e, [hl]
+	inc hl
+	ld d, [hl]
+	ld a, e
+	or d
+	jr z, .psbtn_reset
+	dec de
+	ld [hl], d
+	dec hl
+	ld [hl], e
+	call GetJoypad
+	ld hl, hJoyState	; hJoyState = $FFA3
+	ld a, [hl]
+	and $46				; UP + B + SELECT brings you to the SRAM clear screen.
+	cp $46
+	jr z, .psbtn_sramclear
+	ld a, [hl]
+	and $04				; SELECT will bring you to the debug menu.
+	jr nz, .psbtn_gotodebug
+	ld a, [hl]
+	and $09
+	ret z
+	
+.psbtn_play
+	ld a, $00			;MainMenu
+	jr .psbtn_nextseq
+
+.psbtn_gotodebug
+if DEBUG 
+	ld a, $01			;DebugMenu
+	jr .psbtn_nextseq
+else
+	ret
+endc
+
+.psbtn_sramclear
+	ld a, $02			;SRAMClearMenu
+	
+.psbtn_nextseq
+	ld [wSequence + 1], a	; wSequence + 1 = $CB5F
+	ld hl, wSequence	; wSequence = $CB5E
+	set 7, [hl]
+	ret
+
+.psbtn_reset
+	ld hl, wSequence	; wSequence = $CB5E
+	inc [hl]
+	xor a
+	ld [wNextBGM], a ;c1a7
+	ld [wNextBGM + 1], a ;c1a8
+	ld hl, wSoundFade ;c1a5
+	ld [hl], $08
+	ret	
+	
+FadeMusicOut:: ;615c
+	ld a, [wSoundFade] ;c1a5
+	and a
+	ret nz
+	ld a, $03
+	ld [wSequence + 1], a
+	ld hl, wSequence
+	set 7, [hl]
+	ret	
+
+SetLYOverrides:: ;616c
+	ld hl, wLYOverrides
+	ld c, $30
+.setly_loop
+	ldi [hl], a
+	dec c
+	jr nz, .setly_loop
+	ret
+
+PrintPMSubtitle:: ;6176
+	;ld hl, _RAM_C31A_ ;c31a
+	coord hl, 2, 6
+	ld b, $0F
+	ld a, $69
+	jr LoadPrintArea ;6186
+
+PrintVersion:: ;617f
+	;ld hl, _RAM_C2B8_ ;c2b8
+	coord hl, 4, 1
+	ld b, $09
+	ld a, $60
+	
+LoadPrintArea:: ;6186
+	ldi [hl], a
+	inc a
+	dec b
+	jr nz, LoadPrintArea
+	ret
+
+PrintPMJapaneseChara:: ;618C
+	;ld hl, _RAM_C2D7_
+	coord hl, 15, 2
+	ld a, $BA
+	ld bc, $0404
+	jr PrintBoxArea
+
+PrintPokemonLogo:: ;6196
+	;ld hl, _RAM_C2EB_
+	coord hl, 15, 3
+	ld [hl], $B8
+	;ld hl, _RAM_C2FF_
+	coord hl, 15, 4
+	ld [hl], $B9
+	;ld hl, _RAM_C2C9_
+	coord hl, 1, 2
+	ld a, $80
+	ld bc, $0E04
+	
+PrintBoxArea::
+	ld de, $0014
+	push bc
+	push hl
+.xloop
+	ldi [hl], a
+	inc a
+	dec b
+	jr nz, .xloop
+	pop hl
+	add hl, de
+	pop bc
+	dec c
+	jr nz, PrintBoxArea
+	ret
+	
+PrintCopyRight::
+	;ld hl, _RAM_C3F7_
+	coord hl, 3, 17
+	ld a, $41
+	ld b, $0D
+.copyright_loop
+	ldi [hl], a
+	inc a
+	dec b
+	jr nz, .copyright_loop
+	ret
+	
+SRAMClearMenu:: ;61c6
+	call ClearTileMap
+	call GetMemSGBLayout
+	call LoadFont
+	call LoadFontExtra
+	ld hl, Window_SRAMClearPOS
+	call PrintStringIntoMessageWindow ;e70
+	ld hl, Text_SRAMClearChoice
+	call SetNormalSelectMenuText ;1d50
+	call DrawNormalSelectMenu ;1d87
+	jp c, Init
+	ld a, [wMenuCursorY]	; wMenuCursorY = $CC2A
+	cp $01
+	jp z, Init
+	ld hl, InitAllSRAMBanks	; 143B6
+	ld a, $05
+	call FarCall_hl
+	jp Init
+
+Window_SRAMClearPOS:: ;617f
+db $00, $BD, $3D, $C3, $C9, $7F, $8D, $E3, $1B, $12, $E3, $8F, $83, $D8, $80, $DD
+db $4F, $87, $D8, $80, $7F, $BC, $CF, $BD, $B6, $E6, $57
+
+Text_SRAMClearChoice:: ;6212
+db $00, $07, $0E, $0B, $13, $1A, $62, $01, $C0, $02, $B2, $B2, $B4, $50, $CA, $B2
+db $50	
```

Comment:
This should be changed to represent the Gen 2 VerticalMenu Menu Header format (Documentation at [pokecrystal/docs/menu.md](https://github.com/pret/pokecrystal/blob/725148443f0ed0f70af747259ef49643359e92a2/docs/menu.md#verticalmenu)).